### ISSUE TITLE
SVM using predict_proba gives wrong classification when number of samples per class is 1.

### ISSUE THREAD

**Issue by jaskaran1** [cmusatyalab/openface#250]
Hi @bamos ,

**Context of the issue:**
I tested openface using 1 sample per class and used the linear kernel SVM and get misclassifications.

**Current Behaviour:**
For example
Labels: A,B
Feature:[1,2,3],[7,20,30]
Prediction:B,A [using svm.predict_proba() logic]

**Expected Behaviour:**
Prediction:A,B [using svm.predict() logic]

**Reason and how to duplicate:**
This is because I get  classification probabilities from predict_proba which are inconsistent with the svm.predict() function . 

On duplicating the sample per class, I get the right probabilities from predict_proba.

On further investigation, I found that this is a problem with sklearn's SVM predict_proba.

Please go through this  [link](http://stackoverflow.com/questions/17017882/scikit-learn-predict-proba-gives-wrong-answers).

The reason for this is in [link](http://stackoverflow.com/questions/15111408/how-does-sklearn-svm-svcs-function-predict-proba-work-internally)

I have modified the code of the classifier.py to get the correct classification with 1 sample per class and its working on 1 sample per class which was not working earlier.

I can send a PR if you want.

**Comment by RodrigoPetter** [comment]
@jaskaran1 do you think this https://github.com/cmusatyalab/openface/issues/234#issuecomment-294525287 is telated some way?

**Comment by jaskaran1** [comment]
@RodrigoPetter  
Yes. It is the exact same problem I faced.

**Comment by objch** [comment]
Hi @jaskaran1 would you mind adding a PR or fork containing your fix pls? I'd like to see how you were able to get this working with one sample per class.
